lib/client/doc.js

```diff
@@ -760,6 +769,12 @@ Doc.prototype._pushOp = function(op, callback) {
   op.type = this.type;
   op.callbacks = [callback];
   this.pendingOps.push(op);
+
+  if (op.del) this._heldOps = {};
+  for (var id in this._heldOps) {
+    var transformErr = transformX(op, this._heldOps[id]);
```

Comment:
Again, my understanding of `transformX` isn't great. It's going to mutate `op`. But we're at the end of the function, so I assume that's okay?